src/Token/Parser.php

```diff
@@ -90,7 +91,30 @@ private function parseHeader(string $data): array
      */
     private function parseClaims(string $data): array
     {
-        return (array) $this->decoder->jsonDecode($this->decoder->base64UrlDecode($data));
+        $claims = (array) $this->decoder->jsonDecode($this->decoder->base64UrlDecode($data));
+
+        if (isset($claims[RegisteredClaims::AUDIENCE])) {
+            $claims[RegisteredClaims::AUDIENCE] = (array) $claims[RegisteredClaims::AUDIENCE];
+        }
+
+        foreach (RegisteredClaims::DATE_CLAIMS as $claim) {
+            if (!isset($claims[$claim])) {
+                continue;
+            }
+
+            $claims[$claim] = $this->convertDate((string) $claims[$claim]);
+        }
+
+        return $claims;
+    }
+
+    private function convertDate(string $value): DateTimeImmutable
+    {
+        if (strpos($value, '.') === false) {
+            $value .= '.0';
+        }
+
+        return DateTimeImmutable::createFromFormat('U.u', $value);
```

Comment:
I'd use two different expressions for when a microtime is set or not: this way you avoid overwriting `$value`